src/main/java/org/traccar/web/server/reports/ReportGraph.java

```diff
@@ -0,0 +1,185 @@
+                " WHERE p.device IN :selectedDevices AND p.time BETWEEN :from AND :to ORDER BY p.time", Position.class)
+                .setParameter("selectedDevices", getDevices(report))
+                .setParameter("from", report.getFromDate())
+                .setParameter("to", report.getToDate())
+                .getResultList();
+
+
+        panelStart();
+
+        // heading
+        panelHeadingStart();
+        text("Graph ...");
+        panelHeadingEnd();
+
+        // body
+        panelBodyStart();
+        // period
+        paragraphStart();
+        bold(message("timePeriod") + ": ");
+        text(formatDate(report.getFromDate()) + " - " + formatDate(report.getToDate()));
+        paragraphEnd();
+
+        try {
+            addGraph(getDevices(report), positions);
+        } catch (TraccarException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+
+
+        panelBodyEnd();
+
+        panelEnd();
+
+
+    }
+
+    static class Data {
+        final boolean idle;
+
+        Position start;
+        Position end;
+        double topSpeed;
+        double totalSpeed;
+        double distance;
+        int positionsWithSpeed;
+
+        Data(boolean idle, Position start) {
+            this.start = start;
+            this.idle = idle;
+        }
+
+        long getDuration() {
+            return end.getTime().getTime() - start.getTime().getTime();
+        }
+
+        double getAverageSpeed() {
+            return totalSpeed / positionsWithSpeed;
+        }
+    }
+
+
+
+
+    public void addGraph(List<Device> list, List<Position> positions) throws TraccarException{
+
+        text("<div id=\"graphdiv\" style=\"width:1100px; height:400px;\"></div>");
+        text("<script type=\"text/javascript\">");
+        text("g = new Dygraph(");
+
+        // containing div
+        text("document.getElementById(\"graphdiv\"),");
+
+        text("\"");
+        text(getData(list, positions));
+        text("\"");
+
+        text(","
+                + "{"
+                + "   connectSeparatedPoints: true,"
+                + "   showRangeSelector: true"
+                + " }"
+                + " );");
+        text("</script>");
+    }
+
+
+
+    String getData(List<Device> listDevices, List<Position> positions) throws TraccarException {
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+
+        String output="time";
+
+
+        Map<Long, List<String>> keysMap = new HashMap<>();
+        Map<String, String> deviceSensorMap = new HashMap<>();
+        Map<String, Integer> deviceSensorOrderMap = new HashMap<>();
+System.out.println("broj uredjaja"+listDevices.size());
+        for (Device device: listDevices) {
+            List<String> keys = new ArrayList<String>();
+System.out.println("broj senzora"+device.getName()+" "+device.getSensors());
+
+            if(device.getSensors()==null)
```

Comment:
Don't use `if` statement without following block of code in curly braces. This is considered a bad practice.